crates/hulk_replayer/src/main.rs

```diff
@@ -0,0 +1,196 @@
+        _positions: Joints<f32>,
+        _stiffnesses: Joints<f32>,
+        _leds: Leds,
+    ) -> Result<()> {
+        Ok(())
+    }
+}
+
+impl CameraInterface for ReplayerHardwareInterface {
+    fn read_from_camera(&self, _camera_position: CameraPosition) -> Result<YCbCr422Image> {
+        panic!("Replayer cannot produce data from hardware")
+    }
+}
+
+impl IdInterface for ReplayerHardwareInterface {
+    fn get_ids(&self) -> Ids {
+        self.ids.clone()
+    }
+}
+
+impl MicrophoneInterface for ReplayerHardwareInterface {
+    fn read_from_microphones(&self) -> Result<Samples> {
+        panic!("Replayer cannot produce data from hardware")
+    }
+}
+
+impl NetworkInterface for ReplayerHardwareInterface {
+    fn read_from_network(&self) -> Result<IncomingMessage> {
+        panic!("Replayer cannot produce data from hardware")
+    }
+    fn write_to_network(&self, _message: OutgoingMessage) -> Result<()> {
+        Ok(())
+    }
+}
+
+impl PathsInterface for ReplayerHardwareInterface {
+    fn get_paths(&self) -> Paths {
+        Paths {
+            motions: "etc/motions".into(),
+            neural_networks: "etc/neural_networks".into(),
+            sounds: "etc/sounds".into(),
+        }
+    }
+}
+
+impl RecordingInterface for ReplayerHardwareInterface {
+    fn should_record(&self) -> bool {
+        false
+    }
+    fn set_whether_to_record(&self, _enable: bool) {}
+}
+
+impl SensorInterface for ReplayerHardwareInterface {
+    fn read_from_sensors(&self) -> Result<SensorData> {
+        panic!("Replayer cannot produce data from hardware")
+    }
+}
+
+impl SpeakerInterface for ReplayerHardwareInterface {
+    fn write_to_speakers(&self, _request: SpeakerRequest) {}
+}
+
+impl TimeInterface for ReplayerHardwareInterface {
+    fn get_now(&self) -> SystemTime {
+        SystemTime::now()
+    }
+}
+
+impl HardwareInterface for ReplayerHardwareInterface {}
+
+fn main() -> Result<()> {
+    install()?;
+    let framework_parameters_path = args()
+        .nth(1)
+        .unwrap_or("etc/parameters/framework.json".to_string());
+    let keep_running = CancellationToken::new();
+    set_handler({
+        let keep_running = keep_running.clone();
+        move || {
+            keep_running.cancel();
+        }
+    })?;
+
+    let file =
+        File::open(framework_parameters_path).wrap_err("failed to open framework parameters")?;
+    let framework_parameters: FrameworkParameters =
+        from_reader(file).wrap_err("failed to parse framework parameters")?;
+
+    let hardware_interface = ReplayerHardwareInterface {
+        ids: Ids {
+            body_id: "webots".to_string(),
+            head_id: "webots".to_string(),
+        },
+    };
+
+    let ids = hardware_interface.get_ids();
+
+    let mut replayer = Replayer::new(
+        Arc::new(hardware_interface),
+        framework_parameters.communication_addresses,
+        framework_parameters.parameters_directory,
+        ids.body_id,
+        ids.head_id,
```

Comment:
Yes, correct, we could improve that. I also thought to include a snapshot of the configuration parameters during recording. We could make this more sophisticated, but would it be fine to do that in the next iteration?

Reply:
This would already be completely broken for tests in the lab, as the `webots` parameters have considerably different field dimensions.
I'd strongly advise against knowingly running replay with incorrect parameters. An unreliable debug tool may be worse than a not having one.
If you want to refrain from a sophisticated solution in this iteration, maybe we can find a simple one for this iteration?

Reply:
Ok, I see. How about passing them as command line arguments to the replayer command?

Reply:
I guess having them as command line arguments is fine for this iteration. You'd have to manually rauspopeln them from the `hardware_ids.json`, but that's fine for now I think.

Reply:
Implemented now a very simple approach to get the arguments. Since it will probably be improved soon :tm:, I decided against `clap`.

Reply:
Damn, fixed :man_facepalming: